gettsim/benefits/alg2.py

```diff
@@ -221,58 +221,51 @@ def grossinc_alg2(household):
     )
 
 
-def einkommensanrechnungsfrei(household, params):
-    """Determine the amount of income that is not deducted. Varies withdrawal rates
-    depending on monthly earnings."""
-    # Calculate the amount of children below the age of 18.
-    child0_18_num = (household["child"] & household["age"].between(0, 18)).sum()
-
-    # 100€ is always 'free'
-    household.loc[(household["m_wage"] <= params["a2grf"]), "ekanrefrei"] = household[
-        "m_wage"
-    ]
-    # until 1000€, you may keep 20% (withdrawal rate: 80%)
-    household.loc[
-        (household["m_wage"].between(params["a2grf"], params["a2eg1"])), "ekanrefrei",
-    ] = params["a2grf"] + params["a2an1"] * (household["m_wage"] - params["a2grf"])
-    # from 1000 to 1200 €, you may keep only 10%
-    household.loc[
-        (household["m_wage"].between(params["a2eg1"], params["a2eg2"]))
-        & (child0_18_num == 0),
-        "ekanrefrei",
-    ] = (
-        params["a2grf"]
-        + params["a2an1"] * (params["a2eg1"] - params["a2grf"])
-        + params["a2an2"] * (household["m_wage"] - params["a2eg1"])
-    )
-    # If you have kids, this range goes until 1500 €,
-    household.loc[
-        (household["m_wage"].between(params["a2eg1"], params["a2eg3"]))
-        & (child0_18_num > 0),
-        "ekanrefrei",
-    ] = (
-        params["a2grf"]
-        + params["a2an1"] * (params["a2eg1"] - params["a2grf"])
-        + params["a2an2"] * (household["m_wage"] - params["a2eg1"])
-    )
-    # beyond 1200/1500€, you can't keep anything.
-    household.loc[
-        (household["m_wage"] > params["a2eg2"]) & (child0_18_num == 0), "ekanrefrei",
-    ] = (
-        params["a2grf"]
-        + params["a2an1"] * (params["a2eg1"] - params["a2grf"])
-        + params["a2an2"] * (params["a2eg2"] - params["a2eg1"])
-    )
-    household.loc[
-        (household["m_wage"] > params["a2eg3"]) & (child0_18_num > 0), "ekanrefrei",
-    ] = (
-        params["a2grf"]
-        + params["a2an1"] * (params["a2eg1"] - params["a2grf"])
-        + params["a2an2"] * (params["a2eg3"] - params["a2eg1"])
-    )
-    # Children income is fully deducted, except for the first 100 €.
-    household.loc[(household["child"]), "ekanrefrei"] = np.maximum(
-        0, household["m_wage"] - 100
+def e_anr_frei(household, params):
+    """Calculate income not subject to transfer withdrawal for the household.
+
+    Determine the gross income that is not deducted. Withdrawal rates depend
+    on monthly earnings and on the number of children in the household. § 30 SGB
+    II. Since 01.04.2011 § 11b."""
+
+    # Calculate the number of children below the age of 18.
+    num_childs_0_18 = (household["child"] & household["age"].between(0, 18)).sum()
```

Comment:
does `.between(0, 18)` include 18? The law is
>  die entweder mit mindestens einem minderjährigen Kind in Bedarfsgemeinschaft leben oder die mindestens ein minderjähriges Kind haben

cf https://www.buzer.de/gesetz/2602/a170599.htm

Reply:
It is `<=`. We need minderjährig? Then I change to `.between(0, 17)`

Reply:
Best `< 18`. Who knows if we ever work at a monthly level and end up with fractions.

Reply:
Done